File: CyFile-Android/app/src/test/java/at/tugraz/tc/cyfile/secret/HashedSecretUnitTest.java
package at.tugraz.tc.cyfile.secret;

import org.junit.Assert;
import org.junit.Test;

public class HashedSecretUnitTest extends BaseUnitTest {

    @Test
    public void bothConstructorsShouldGiveEqualValue() {
        HashedSecret hashedSecret = new HashedSecret(() -> "111222", new ApacheCodecBase64());
        Secret pinPatternSecret = new PinPatternSecret("111222");
        Assert.assertTrue(hashedSecret.equals(pinPatternSecret));
    }

}

File: CyFile-Android/app/src/test/kotlin/at/tugraz/tc/cyfile/secret/HashedSecretUnitTest.kt
package at.tugraz.tc.cyfile.secret

import at.tugraz.tc.cyfile.BaseUnitTest
import at.tugraz.tc.cyfile.crypto.impl.ApacheCodecBase64
import at.tugraz.tc.cyfile.secret.impl.HashedSecret
import at.tugraz.tc.cyfile.secret.impl.PinPatternSecret
import org.junit.Assert
import org.junit.Ignore
import org.junit.Test

class HashedSecretUnitTest : BaseUnitTest() {

    @Test
    @Ignore
    fun bothConstructorsShouldGiveEqualValue() {
        val hashedSecret = HashedSecret(PinPatternSecret("111222"), ApacheCodecBase64())
        val pinPatternSecret = PinPatternSecret("111222")
        Assert.assertTrue(hashedSecret.secretValue == pinPatternSecret.secretValue)
    }

}
